Library/Homebrew/download_strategy.rb

```diff
@@ -388,6 +388,25 @@ def _fetch
   end
 end
 
+# Will convert "https://downloads.sourceforge.net/project/..." URLs to mirrorservice.org
+# In case environment variable HOMEBREW_NO_INSECURE_REDIRECT is set, so the Sourceforge is not ultimately broken
+class CurlSourceForgeSecureMirrorDownloadStrategy < CurlDownloadStrategy
+  def fetch
+    unless ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+      ohai "Sourceforge HTTPS: #{@url}"
+      @url = https_mirror
+      ohai "=> converted to #{@url}"
+    end
+    super
+  end
```

Comment:
How about:
```ruby
def fetch
  mirrors.unshift *https_mirrors
  super
end
```

To make it work, you need to add: 
```ruby
class CurlDownloadStrategyHTTPSRedirectError < CurlDownloadStrategyError
  def to_s
    super + "\nHTTPS to HTTP redirect detected & HOMEBREW_NO_INSECURE_REDIRECT is set."
  end
end
```
in the `exceptions.rb`. And change `raise "HTTPS to HTTP redirect detected & HOMEBREW_NO_INSECURE_REDIRECT is set."` to `raise CurlDownloadStrategyHTTPSRedirectError.new(@url)`.